spec/models/result_spec.rb

```diff
@@ -165,4 +165,41 @@
       end
     end
   end
+
+  describe '.set_release_on_results' do
+    let!(:assignment) { create :assignment_with_criteria_and_results }
+    let!(:released) { create :released_result }
```

Comment:
I don't think this is doing what you think it's doing. The result being created here won't be associated with the `assignment` you created above. Same with `create(:incomplete_result)` below. Instead, it's creating a completely different result.

You can check out what the `assignment_with_criteria_and_results` factory is doing, but I'm guessing it's creating some incomplete results all by itself.